src/database/postgres.go

```diff
@@ -0,0 +1,124 @@
+package database
+
+import (
+	"context"
+	"fmt"
+
+	pb "accretional.com/semantifly/proto/accretional.com/semantifly/proto"
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/pgconn"
+	"google.golang.org/protobuf/encoding/protojson"
+)
+
+type PgxIface interface {
+	Begin(context.Context) (pgx.Tx, error)
+	Close(context.Context) error
+	Query(ctx context.Context, sql string, args ...interface{}) (pgx.Rows, error)
+	Exec(ctx context.Context, sql string, arguments ...interface{}) (pgconn.CommandTag, error)
+}
+
+func initializeDatabaseSchema(ctx context.Context, conn PgxIface) error {
+	tx, err := conn.Begin(ctx)
+	if err != nil {
+		return fmt.Errorf("unable to connect to database: %w", err)
+	}
+	defer tx.Rollback(ctx)
+
+	// Create the main table using protobuf
+	_, err = tx.Exec(ctx, `
+		CREATE TABLE IF NOT EXISTS index_list (
+			name TEXT PRIMARY KEY,
+			entry JSONB
+		)
+	`)
+	if err != nil {
+		return fmt.Errorf("failed to create table: %w", err)
+	}
+
+	// Create indexes
```

Comment:
Don't create these indices by default right now, add a separate function like createProtoFieldIndex that takes a param (eg "wordOccurrences") and makes an index for it, or just remove this

Reply:
Done. Moved it to a different function